jdaviz/core/loaders/resolvers/resolver.py

```diff
@@ -498,6 +615,122 @@ def on_observation_select_changed(self, _=None):
+            if self.parsed_input_is_query and self.treat_table_as_query:
+                self._display_observation_footprints()
+            super().toggle_custom_toolbar()
+        else:
+            super().toggle_custom_toolbar()
+            if self.parsed_input_is_query and self.treat_table_as_query:
+                self._remove_observation_footprints()
+
+    def _on_region_select(self, msg):
+        """Handle footprint click events."""
+        if not self._footprint_marks:
+            return
+        if not (self.parsed_input_is_query and self.treat_table_as_query):
+            return
+
+        click_x, click_y = msg.x, msg.y
+        selected_idx = find_closest_polygon_mark(click_x, click_y, self._footprint_marks)
+
+        if selected_idx is not None:
+            # Get selected rows from the table
+            currently_selected = set()
+            if hasattr(self, 'observation_table') and self.observation_table is not None:
+                for row in self.observation_table.selected_rows:
+                    idx = self.observation_table.items.index(row)
+                    currently_selected.add(idx)
+
+            # Toggle behavior
+            if selected_idx in currently_selected:
+                currently_selected.discard(selected_idx)
+                for mark in self._footprint_groups.get(selected_idx, []):
+                    mark.selected = False
+            else:
+                currently_selected.add(selected_idx)
+                if selected_idx in self._footprint_groups:
+                    for mark in self._footprint_groups[selected_idx]:
+                        mark.selected = True
+
+            # Update the table selection
+            if hasattr(self, 'observation_table') and self.observation_table is not None:
+                if currently_selected:
+                    self.observation_table.select_rows(sorted(list(currently_selected)))
+                else:
+                    # Clear selection
+                    self.observation_table.selected_rows = []
+
+    def _display_observation_footprints(self):
+        if 's_region' not in self.observation_table.headers_avail:
+            return
+
+        viewer = self.app._jdaviz_helper.default_viewer._obj.glue_viewer
+        wcs = getattr(viewer.state.reference_data, 'coords', None)
+
+        successfully_displayed = 0
+        new_marks = []
+
+        for idx, row in enumerate(self.observation_table.items):
+            s_region_str = row.get('s_region', '')
+
+            if not s_region_str or (isinstance(s_region_str, str) and s_region_str.strip() == ''):
+                continue
+            if not is_stcs_string(s_region_str):
+                continue
+
+            region = stcs_string2region(s_region_str)
+            pixel_region = region.to_pixel(wcs)
+            x, y = pixel_region.vertices.x, pixel_region.vertices.y
+
+            mark = RegionOverlay(
+                viewer,
+                f"ObsTable_{idx}",
```

Comment:
This argument is intended to be a `PolygonSkyRegion`. If you use the `viewer._add_region_overlay` method, that's handled for you.